### ISSUE TITLE
pennyRound option

### ISSUE THREAD

**Issue by arecker** [arecker/Daisy#3]
Ability to pass in an option to round up or down on penny fractions

``` javascript
daisy('1.463', { pennyRound: 'up'}).equals() // '1.47'
daisy('1.463', { pennyRound: 'down'}).equals() // '1.46'
```

**Comment by arecker** [comment]
This will be the last enhancement before tagging a version

**Comment by bignimbus** [comment]
I'd like to take this one.

**Comment by arecker** [comment]
@bignimbus have at it!  Thanks for helping

**Comment by bignimbus** [comment]
What about something like this:

``` js
{
"format": "n.nn",
"commas": true,
"INR": true,
"truncate": true
}
```

Where `format` (default: "n.nn") is a string containing the number of place values to print, `commas` (default: false) indicate whether to print large numbers with commas (e.g. 1,000.00), `INR` (default: false) indicates whether to use commas peculiar to Indian currencies, and `truncate` (default: false) indicates to use `Math.floor` instead of `Math.round`.

**Comment by bignimbus** [comment]
`"format"` would then give us flexibility to round to higher values:

`"format": "n000"` would mean to round/truncate to the nearest 1e3.  Useful for apps that use natural language.

**Comment by arecker** [comment]
I really like the `format` idea.  That would probably allow us to do away with the silly `printDollarSign` option I made (could just use `"format": " "$n.nn"`).  I'm OK with that.  Same with `commas` and `INR`.  Love it.

I'm still not totally pumped about `truncate`.  Rather than a `true` or `false` for simply runding off leftovers, I wanted to provide an option to round either up or down.  Underneath, it would probably use, respectively, `Math.cieling` and `Math.floor` under the covers.  I've never worked in finance so I'm not sure what is normally done with penny fractions, but I'd figure lets go with more options over less.

Does that make sense?

**Comment by bignimbus** [comment]
That's fair.  If the user leaves `"round"` blank, do we default to `Math.round()`?

**Comment by arecker** [comment]
Yep!  It should probably just default on natural rounding - i.e. down if less than `5` and up if greater than or equal to `5`

**Comment by bignimbus** [comment]
<img src="http://i.imgur.com/Ie17FZ6.gif">

**Comment by arecker** [comment]
Jasmine Tests:

``` javascript
describe('pennyRound option', function(){
    it('should round up when "up"', function(){
        var actual = daisy('12.322', { pennyRound: 'up' }).equals();
        expect(actual).toBe('12.33');
    }
    it('should round down when "down"', function(){
        var actual = daisy('12.329', { pennyRound: 'down' }).equals();
        expect(actual).toBe('12.32');
    }
}
```

**Comment by bignimbus** [comment]
My thought was that using the `format` option would be an elegant way of combining configurations.  If we're adding an explicit `round` method and `pennyRound` option, `format` really doesn't have a place anymore.  Thoughts?

**Comment by arecker** [comment]
I agree. Don't see a place for it. Sorry I wasn't more clear before - just
had to think out loud I guess.

**Comment by arecker** [comment]
I went ahead and added the `round` representation as a private method.  I've set it up so this option can just re use it.
